auditmiddleware/_api.py

```diff
@@ -682,7 +687,20 @@ def _create_target_resource(self, target_project, res_spec, res_id,
                     payload,
                     res_spec)
 
-        type_uri = res_spec.el_type_uri if rid else res_spec.type_uri
+        type_uri = None
+        if rid:
+            # if we have a resource ID (from path or payload), use element URI
+            type_uri = res_spec.el_type_uri or res_spec.type_uri
+        elif request and request.method == 'POST' and not res_spec.singleton:
+            # if it's a POST to a non-singleton resource
+            # (even if failed, so no rid yet)
+            # the intent is an element, so use element URI
+            type_uri = res_spec.el_type_uri or res_spec.type_uri
+
+        # Fallback: If no specific logic applied, or if it's a singleton
+        if type_uri is None:
+            type_uri = res_spec.type_uri
```

Comment:
You could set it to `res_spec.type_uri` by default instead of to `None`.

Reply:
Great shout, done.